pkg/otlp/collector.go

```diff
@@ -0,0 +1,166 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+package otlp
```

Comment:
Any advantage to put OTLP behind a build flag ? Will this works on all agent platform ?

Reply:
This will work on all platforms; the OpenTelemetry Collector uses these packages and they provide support for Windows, macOS and Linux

Reply:
Depending on the size of the OTLP package is could be interesting to not compile with it for the serverless or dogstatsd agent.

Reply:
https://github.com/knz/go-binsize-viz says the current size is 10k (0.022%) of the Agent size, so it should be fine fo rnow. We can revisit in the future if this becomes dramatically bigger